scala-vector-tile/src/main/scala/geotrellis/vectortile/Commands.scala

```diff
@@ -0,0 +1,129 @@
+package geotrellis.vectortile
+
+import collection.mutable.ListBuffer
+
+import geotrellis.vector._
+
+object Command {
+
+    val MoveTo: Int = 1
+
+    val LineTo: Int = 2
+
+    val ClosePath: Int = 7
+
+    type GeomType = vector_tile.Tile.GeomType.EnumVal
+    val POINT = vector_tile.Tile.GeomType.POINT
+    val LINESTRING = vector_tile.Tile.GeomType.LINESTRING
+    val POLYGON = vector_tile.Tile.GeomType.POLYGON
+
+    def parse(geomType: GeomType, extent: Int, commands: Seq[Int]): Geometry = {
+
+        val scale: Double = extent / 256.0
+
+        val point_lists: ListBuffer[List[(Double, Double)]] =
+            ListBuffer.empty[List[(Double, Double)]]
+
+        interpret_commands()
+
+        return geomType match {
+
+            case POINT =>
+                if (point_lists.length == 1) {
+                    Point(point_lists.head.head)
+                } else {
+                    MultiPoint(point_lists.map(
+                        (pt: List[(Double, Double)]) => Point(pt.head)))
+                }
+
+            case LINESTRING =>
+                if (point_lists.length == 1) {
+                    Line(point_lists.head)
+                } else {
+                    MultiLine(point_lists.map(
+                        (ln: List[(Double, Double)]) => Line(ln)))
+                }
+
+            case POLYGON =>
+                if (point_lists.length == 1) {
+                    Polygon(point_lists.head)
+                } else {
+                    MultiPolygon(point_lists.map(
+                        (pg: List[(Double, Double)]) => Polygon(pg)))
+                }
+
+            case _ =>
+                // this should probably be logged.
+                null
```

Comment:
Shouldn't return null, if it's an exceptional case then we should throw an exception